### DISCUSSION TITLE
Motifs in multiple time series

### DISCUSSION THREAD

**Discussion by tomnewg** [TDAmeritrade/stumpy#458] [Help: Coding & Implementations]
Hi stumpy community,

I was wondering if someone can point me in the right direction. I have been experimenting with the Matrix Profile and stumpy. Such an awesome new methodlogy. In my current analysis, I have multiple time series where I want to find common Motifs. 
I looked at the https://stumpy.readthedocs.io/en/latest/Tutorial_AB_Joins.html
Is there a way to run this analysis over more than two time series? Is it possible that I can search for a specific Motif that I found in one time series to find in another?

Many thanks for your help

Tom

**Comment by seanlaw** [comment]
@tomnewg Welcome to the STUMPY community and thank you for your question!

So, if you already have a motif in mind and you want to search for a specific pattern then you may be interested this [fast pattern matching](https://stumpy.readthedocs.io/en/latest/Tutorial_Pattern_Matching.html) tutorial. You may need to wrap this into a simple `for-loop` if you want to loop through multiple time series.

Additionally, if you want to find motifs across multiple time series (i.e., you don't have a pattern in mind) then you might try the "ostinato" algorithm and you can learn more from [this tutorial](https://stumpy.readthedocs.io/en/latest/Tutorial_Consensus_Motif.html).

If this doesn't quite answer your question then please feel free to provide some more information or a minimum reproducible code sample and I can try to help.

**Reply by tomnewg** [reply]
Hi Sean,

thank you so much for your answers. I will give this a go and will revert back.
Thank you so much for stumpy in general and your youtube video you made for the PyData Global conference. I stumbled over Matrix Profile and all the papers and was thinking if there were a python library that applies these papers that would be a dream. And then I found your video and you did exactly that. Awesome work!
